src/util/nondet_bool.h

```diff
@@ -15,14 +15,17 @@ Author: Chris Smowton, chris@smowton.net
 #include "std_expr.h"
 #include "std_types.h"
 
-/// \par parameters: Desired type (C_bool or plain bool)
+/// \param type desired type (C_bool or plain bool)
+/// \param source_location source location
 /// \return nondet expr of that type
-inline exprt get_nondet_bool(const typet &type)
+inline exprt get_nondet_bool(
+  const typet &type,
+  const source_locationt &source_location = source_locationt())
```

Comment:
Let's not make this optional: there is a single call site (and the next commit adds a second one), which could trivially pass in a proper source location.